src/main/java/org/jabref/logic/util/io/CiteKeyBasedFileFinder.java

```diff
@@ -93,8 +94,8 @@ private boolean matches(String filename, String citeKey) {
             if (Files.exists(directory)) {
                 try (Stream<Path> files = Files.find(directory, Integer.MAX_VALUE, isFileWithCorrectExtension)) {
                     result.addAll(files.collect(Collectors.toSet()));
-                } catch (IOException e) {
-                    LOGGER.error("Problem in finding files", e);
+                } catch (UncheckedIOException e) {
```

Comment:
That's interesting, why/when does this Exception occur?

Reply:
To be honest, I have no idea. The documentation babbles something about "a late exception that is thrown when the stream is finished". I guess the file is found, added to the stream but when a bit later `toSet` is called it is no longer there. I also don't understand why it has to be an unchecked exception instead of a normal `IOException`.

Reply:
I refactored the code slightly to get rid of these exception handling in streams.
I think this is should work better now because the stream is no longer directly closed and then accessed. But I could be mistaken.

Reply:
I don't think that works (see below). But thanks for the other fix, that's a nice improvement over my hacky code.